runtime/tests/Test_OpenQasmDevice.cpp

```diff
@@ -69,9 +71,49 @@ TEST_CASE("Test BraketRunner::runCircuit()", "[openqasm]")
 
     auto &&circuit = builder.toOpenQasm();
 
+    // Initializing the Python interpreter is required to run the circuit.
+    // We use pybind11 for this since nanobind has no intention to support embedding a Python
+    // interpreter in C++.
+    if (!Py_IsInitialized()) {
+        pybind11::initialize_interpreter();
+    }
+
     OpenQasm::BraketRunner runner{};
-    auto &&results = runner.runCircuit(circuit, "default", 100);
-    CHECK(results.find("GateModelQuantumTaskResult") != std::string::npos);
+
+    SECTION("Test BraketRunner::runCircuit()")
+    {
+        auto &&results = runner.runCircuit(circuit, "default", 100);
+        CHECK(results.find("GateModelQuantumTaskResult") != std::string::npos);
+    }
+
+    SECTION("Test BraketRunner::Probs()")
+    {
+        auto &&probs = runner.Probs(circuit, "default", 100, 2);
+        CHECK(probs.size() == 4); // For a 2-qubit system
+        CHECK((probs[0] >= 0.0 && probs[0] <= 1.0));
+        CHECK((probs[1] >= 0.0 && probs[1] <= 1.0));
+        CHECK((probs[2] >= 0.0 && probs[2] <= 1.0));
+        CHECK((probs[3] >= 0.0 && probs[3] <= 1.0));
+    }
+
+    SECTION("Test BraketRunner::Sample()")
+    {
+        auto &&samples = runner.Sample(circuit, "default", 100, 2);
+        CHECK(samples.size() == 200); // Expecting 100 * 2 = 200 samples
+        for (const auto &sample : samples) {
+            REQUIRE(
+                (sample == 0 || sample == 1)); // Each sample should be 0 or 1 for a 2-qubit state
+        }
+    }
+
+    // TODO: These operations are not yet functional
```

Comment:
Still working on this one?

Reply:
I'm not sure either. @maliasadi ?

Reply:
@joeycarter There aren't any measurement checks in these two functions as far as I can tell. As a result, users can send malformed circuits to these functions and still expect some results! We need to also validate the MP type before `device.run` calls in `py_run_circuit`.

Reply:
@dime10 and @maliasadi I've finished adding tests for `Expval` and `Var`. The trick was to add measurement processes to the OpenQASM program like so:

```cpp
builder.toOpenQasmWithCustomInstructions("#pragma braket result variance y(q[0])");
```

This way, the `values` attribute of the run results get populated with the expectation value / variance. I've also added some simple error checking that raises a `RuntimeError` if we've called `Expval` or `Var` and not specified a measurement process (resulting in the `values` attribute being an empty list).